### PR TITLE
fix all JSDoc eslint errors/warnings across the entire repository

### PR DESCRIPTION
**Pull request by dsifford**
Extracted from #16869

## Description

This PR simply fixes any eslint errors/warnings that arose as a result of switching to `eslint-plugin-jsdoc`.

## How has this been tested?

ESlint runs and there are no longer any warnings associated with the changes to JSDoc-related lints.

## Screenshots 

N/A

## Types of changes

Code quality

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 

**Note:** This is my first cherry-picked PR so hopefully I didn't blow anything up!

### CODE REVIEW

File: packages/rich-text/src/split.js [hunk 1]
```diff
@@ -9,9 +9,13 @@ import { replace } from './replace';
  * split at the given separator. This is similar to `String.prototype.split`.
  * Indices are retrieved from the selection if none are provided.
  *
- * @param {Object}        value    Value to modify.
+ * @param {Object}        richText
```
Comment (talldan) [inline packages/rich-text/src/split.js]: I wonder if `richText` is the right way to describe this. It seems to be called `value` everywhere at the moment.

**Review by talldan — APPROVED** [review]
This looks good. We'd probably want to merge it as soon as possible to avoid conflicts. 

I've avoided commenting on any doc formatting that was already incorrect. I only spotted a couple of things that might need some modification.

Looks like the build failed as there are some altered files after running the build, probably some docs that need to be generated using `npm run docs:build`.

File: package-lock.json [hunk 2]
```diff
@@ -4078,7 +4078,25 @@
 				"eslint-plugin-jsx-a11y": "^6.2.1",
 				"eslint-plugin-react": "^7.12.4",
 				"eslint-plugin-react-hooks": "^1.6.0",
+				"globals": "^12.0.0",
```
Comment (gziolo) [inline package-lock.json]: This change shouldn't be part of this PR. I don't see this dependency added to `package.json` file.

**Review by gziolo — COMMENTED** [review]
Great PR, I like all the consistency it brings. I can confirm that `npm run build` is necessary to move forward.

**Review by gziolo — APPROVED** [review]

**Comment by dsifford** [comment]
Fixes have been applied for the issues mentioned. 

I ran `npm run build` to regenerate the docs and what you see in the diff is all that changed for me. Let me know if I missed something.
